### PR TITLE
Remove redundant logic

### PR DESCRIPTION
**Pull request by bratseth**
- Add retry of failing first job to triggerReadyJobs
- Remove triggerFailing as it is now covered by triggerReadyJobs
- Rename BlockedChangeDeployer to ReadyJobsTrigger

@jonmv please review

@mpolden fyi

### CODE REVIEW

File: controller-server/src/main/java/com/yahoo/vespa/hosted/controller/deployment/DeploymentTrigger.java [hunk 1]
```diff
@@ -145,15 +145,24 @@ private void triggerReadyJobs(LockedApplication application) {
...
-        // TODO: How can the first job not be systemTest (second ccondition)?
-        if ( ! jobs.isEmpty() && jobs.get(0).equals(JobType.systemTest) &&
-             application.deploying().get() instanceof Change.VersionChange) {
-            Version target = ((Change.VersionChange)application.deploying().get()).version();
-            JobStatus jobStatus = application.deploymentJobs().jobStatus().get(JobType.systemTest);
-            if (jobStatus == null || ! jobStatus.lastTriggered().isPresent() 
-                || ! jobStatus.lastTriggered().get().version().equals(target)) {
-                application = trigger(JobType.systemTest, application, false, "Upgrade to " + target);
-                controller.applications().store(application);
+        if ( ! jobs.isEmpty() && jobs.get(0).equals(JobType.systemTest) ) {
```
Comment (jonmv) [inline controller-server/src/main/java/com/yahoo/vespa/hosted/controller/deployment/DeploymentTrigger.java]: How can the first job not be system test, though? Isn't the deplyment spec made such that it always is, unless it isEmpty()?

**Review by jonmv — APPROVED** [review]
I think this is great!

So there's no more throttling of retrying failed jobs, but I guess that's ok; failed test / staging jobs, not failing on capacity, are added at the back of the queue, and for production jobs, there may not be a good reason to hold retries? Anyway, it's been like this ever since the ready-trigger was introduced, so I guess it's OK.

File: controller-server/src/main/java/com/yahoo/vespa/hosted/controller/deployment/DeploymentTrigger.java (hunk 1 above)
Comment (bratseth) [reply controller-server/src/main/java/com/yahoo/vespa/hosted/controller/deployment/DeploymentTrigger.java]: Yes, but that code is far away and not explicitly guaranteed in the API, so I don't feel comfortable implicitly assuming it.

Comment (jonmv) [reply controller-server/src/main/java/com/yahoo/vespa/hosted/controller/deployment/DeploymentTrigger.java]: Fair enough.

**Pull request merged**